Collector/Profile.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ *
+ * @internal
+ */
+final class Profile
+{
+    /**
+     * @var string
+     */
+    private $plugin;
+
+    /**
+     * @var string
+     */
+    private $request;
+
+    /**
+     * @var string
+     */
+    private $response;
+
+    /**
+     * @var bool
+     */
+    private $failed = false;
+
+    /**
+     * @return string
+     */
+    public function getPlugin()
+    {
+        return $this->plugin;
+    }
+
+    /**
+     * @param string $plugin
+     */
+    public function setPlugin(string $plugin)
```

Comment:
sorry to only come up with this now, but: do we ever need to change this value? it looks like this and setRequest should be replaced with mandatory constructor arguments. only response and failed are added later.